azurerm/resource_arm_batch_certificate.go

```diff
@@ -0,0 +1,250 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/batch/mgmt/2017-09-01/batch"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmBatchCertificate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmBatchCertificateCreate,
+		Read:   resourceArmBatchCertificateRead,
+		Update: resourceArmBatchCertificateUpdate,
+		Delete: resourceArmBatchCertificateDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"resource_group_name": resourceGroupNameSchema(),
+			"certificate": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(batch.Cer),
+					string(batch.Pfx),
+				}, false),
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Optional:  true, // Required if `format` is "Pfx"
```

Comment:
Could we add a check here to validate this during creation?

Reply:
Added checks in the Create and Update methods after chatting to @tombuildsstuff in Slack